### DISCUSSION TITLE
How to use NocoDB with SQLite and Docker

### DISCUSSION THREAD

**Discussion by mainrs** [nocodb/nocodb#277] [Q&A] [answered]
Hey!

I want to use NocoDB for some local organization and was wondering how to properly set it up using SQLite and Docker. I'd like to run NocoDB inside of Docker but map the folder where the SQLite database is located so I can easily back it up.

However I can't find a reference to the location of the SQLite file anywhere in the repository.

**Comment by pranavxc** [comment, marked as answer]
SQLite file will be generated on runtime(`noco.db`) and it's kept under `/usr/src/app` folder within docker.

**Reply by pranavxc** [reply]
You can mount a local folder with the following docker command 

```
docker run -p 8080:8080 -v /your/local/path:/usr/src/app -name nocodb nocodb/nocodb
```

 _Note : `/usr/src/app` folder contains source code and sqlite database._

<hr>
Or you can copy the existing SQLite file from the docker container using `docker cp` as follows

```
docker cp <container-Id-or-name>:/usr/src/app/noco.db /your/local/path
```


Reference :  

- [docker run --volume](https://docs.docker.com/engine/reference/commandline/run/#mount-volume--v---read-only)
- [docker cp](https://docs.docker.com/engine/reference/commandline/cp/)

**Reply by MarkNBroadhead** [reply]
More recent versions need to mount `/usr/app/data` instead of `/usr/src/app`:

```docker run -p 8080:8080 -v /your/local/path:/usr/app/data --name nocodb nocodb/nocodb```

**Comment by jaanaltosaar** [comment]
Did this end up working? Were you able to use Litestream as well to back up to s3? Thanks @mainrs !

**Reply by mainrs** [reply]
It worked for the file. It's at the path mentioned in the marked answer. I do not know what Litestream is. I simply copy the file.

**Comment by jaanaltosaar** [comment]
Does anyone know how to version control the sqlite database in a nocodb instance?
